bravado/requests_client.py

```diff
@@ -188,28 +129,70 @@ def apply_authentication(self, request):
         return request
 
 
-class SynchronousEventual(object):
-    """An adapter which supports the :class:`crochet.EventualResult` interface
-    for the :class:`SynchronousHttpClient` class.
+def add_response_detail_to_errors(e):
+    """Specific to requests errors. Error detail is not
+    directly visible in `raise_for_status` trace, instead it is
+    located under `e.response.text`
     """
+    if hasattr(e, 'response') and hasattr(e.response, 'text'):
+        # e.args is a tuple, change to list for modifications
+        args = list(e.args)
+        args[0] += (' : ' + e.response.text)
+        e.args = tuple(args)
+    raise e
+
+
+class RequestsResponseAdapter(ResponseLike):
+    """Wraps a requests.models.Response object to provide a uniform interface
+    to the response innards.
+    """
+
+    def __init__(self, requests_lib_response):
+        """
+        :type requests_lib_response: :class:`requests.models.Response`
+        """
+        self._delegate = requests_lib_response
+
+    @property
+    def status_code(self):
+        return self._delegate.status_code
+
+    def json(self, **kwargs):
+        return self._delegate.json(**kwargs)
+
+
+class RequestsFutureAdapter(object):
+    """A future which inputs HTTP params"""
 
     def __init__(self, session, request):
+        """Kicks API call for Requests client
+
+        :param session: session object to use for making the request
+        :param request: dict containing API request parameters
+        """
         self.session = session
         self.request = request
 
-    def result(self, *args, **kwargs):
-        return self.wait(*args, **kwargs)
+    def check_for_exceptions(self, response):
+        try:
+            response.raise_for_status()
+        except Exception as e:
+            add_response_detail_to_errors(e)
 
-    def wait(self, timeout=None):
-        """Perform the request.
+    def result(self, timeout):
+        """Blocking call to wait for API response
 
-        :param timeout: timeout for the request, in seconds
+        :param timeout: timeout in seconds to wait for response
+        :type timeout: integer
+        :return: raw response from the server
+        :rtype: dict
         """
         request = self.request
         log.debug(u"%s %s(%r)", request.method, request.url, request.params)
```

Comment:
Yup, I think even the hostname is usually part of an earlier log line where it establishes the connection for the session, so it's pretty easy to reconcile them.